eyetracking/bids_scripts/things_driftCor.py

```diff
@@ -0,0 +1,664 @@
+import os, glob, sys, json
+import pandas as pd
+import numpy as np
+from scipy.spatial.distance import pdist
+
+from pathlib import Path
+
+import matplotlib
+matplotlib.use('pdf')
+import matplotlib.pyplot as plt
+
+import argparse
+
+from utils import get_list, add_file_paths, driftcorr_fromlast, get_onset_time
+from utils import reset_gaze_time, format_gaze
+
+
+def get_arguments():
+    parser = argparse.ArgumentParser(
+        description='clean up, label, QC and bids-formats the THINGS'
+        'eye-tracking dataset'
+    )
+    parser.add_argument(
+        '--in_path',
+        type=str,
+        required=True,
+        help='absolute path to directory that contains all data (sourcedata)',
+    )
+    parser.add_argument(
+        '--is_final',
+        action='store_true',
+        default=False,
+        help='if true, export drift-corrected gaze into bids format',
+    )
+    parser.add_argument(
+        '--out_path',
+        type=str,
+        required=True,
+        help='absolute path to analysis directory',
+    )
+
+    return parser.parse_args()
+
+
+def get_single_distance(x1, y1, x2, y2, is_distance=False):
+
+    norm_center = (0, 0) if is_distance else (0.5, 0.5)
+
+    pre_x = (x1 - norm_center[0])*1280
+    pre_y = (y1 - norm_center[1])*1024
+    post_x = (x2 - norm_center[0])*1280
+    post_y = (y2 - norm_center[1])*1024
+
+    dist_in_pix = 4164 # in pixels
```

Comment:
I estimated the distance to screen in pixels from the screen dim in pixels and in degrees of visual angle: adjacent = 
 opposite / tan(angle), so that distance = half-screen-size-in-pix/tan(half-screen-size-in-angle). For the width, I got (1280/2)/(math.tan(np.deg2rad(17.5/2))) = 4158.1467, and for the height I got (1024/2)/(math.tan(np.deg2rad(7.0))) = 4169.90537. I averaged the two values and rounded it to 4164 pixels

Reply:
ok, measured distance is 180cm, screen projection width is 55cm, so distance in pixel is 1280/55*180=4189.